*trivial* fixed AboutListTile having an empty icon placeholder when no icon set.

fourman-alex opened the pull request:
## Description

fixed `AboutListTile` having an empty icon placeholder when no icon set.
This is because having AboutListTile.icon defaults to `const Icon(null)` will create an empty `SizedBox` in the underlying `ListTile.leading`



## Related Issues

None

## Tests

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

fourman-alex committed:
fixed AboutListTile having an empty icon placeholder when no icon set.
This is because having AboutListTile.icon default to const Icon(null) will create an empty SizedBox in the underlying ListTile.leading

shihaohong commented:
I tested the change and it does not seem to be making a difference visually. Am I missing something? Also, please add a test to verify that the intended effect is achieved. Thanks!

fourman-alex committed:
added a test for correct title offset of AboutListTile to about_test.dart

fourman-alex committed:
Merge branch 'upstream_master' into about_list_tile_fix_original

fourman-alex commented:
@shihaohong I should have added a test case and a screenshot. Basically if you try running
```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    const ListTile headingTile = ListTile(title: Text('Some Heading'));
    const AboutListTile aboutTile = AboutListTile(child: Text('About'),);

    return MaterialApp(
      home: Scaffold(
        drawer: Drawer(
          child: ListView(
            children: const <Widget>[
              headingTile,
              aboutTile,
            ],
          ),
        ),
        appBar: AppBar(),
      ),
    );
  }
}
```
Before my fix you'd get the following result (the "About" is indented which is the bug)
![Screenshot_1572948340](https://user-images.githubusercontent.com/5362998/68198646-1a83db80-ffc5-11e9-848a-66b9c0ab955d.png)

I added a test to check that the offset of the child in AboutListTile is the same as the title of ListTile

shihaohong reviewed (commented):
Sorry about that, I was looking at `AboutDialog` instead of `AboutListTile`. I have a quick comment about your test case -- I think it can be improved by instead checking its [descendant](https://api.flutter.dev/flutter/flutter_test/CommonFinders/descendant.html) widgets:

```
expect(find.descendant(
  of: find.byType(AboutListTile),
  matching: find.byType(Icon),
), findsNothing);
```

fourman-alex commented:
> Sorry about that, I was looking at `AboutDialog` instead of `AboutListTile`. I have a quick comment about your test case -- I think it can be improved by instead checking its [descendant](https://api.flutter.dev/flutter/flutter_test/CommonFinders/descendant.html) widgets:
> 
> ```
> expect(find.descendant(
>   of: find.byType(AboutListTile),
>   matching: find.byType(Icon),
> ), findsNothing);
> ```

That's a good suggestion. I'll do some changes and update the branch.

fourman-alex committed:
material/about_test.dart: renaming test to clearly state expected result

Co-Authored-By: Shi-Hao Hong <shihaohong@google.com>

fourman-alex committed:
about_test.dart: test "AboutListTile's child should not be offset when the icon is not specified" is now more concise.
removed the drawer animation and ListTile to focus just on AboutListTile not having an icon offset when it was not specified

shihaohong commented on packages/flutter/test/material/about_test.dart:
```diff
@@ -449,6 +449,25 @@ void main() {
     expect(nestedObserver.dialogCount, 1);
   });
+
+  testWidgets(
```
Make this all one line:
```
  testWidgets("AboutListTile's child should not be offset when the icon is not specified.", (WidgetTester tester) async {
```

fourman-alex replied:
Yeah I don't like it either but it's a formatter issue. The problem is this file is hard to work on because you always have to avoid formatting the whole file. What do you think?

shihaohong replied:
We typically don’t work on Flutter source code with dartfmt turned on. I’d just try to stick to Flutter’s style guide thats in the wiki and turn the formatter off

fourman-alex replied:
I guess I missed that. I'll do that. Thank you

shihaohong commented on packages/flutter/test/material/about_test.dart:
```diff
@@ -449,6 +449,25 @@ void main() {
+          ),
+        );
+
+        expect(
```
Formatting here is a little tough to parse:
```
        expect(
          find.descendant(
            of: find.byType(AboutListTile), 
            matching: find.byType(Icon),
          ),
          findsNothing,
        );
```

fourman-alex replied:
My bad. I'll add the commas

fourman-alex replied:
My bad. I'll add the commas

shihaohong commented:
Also, there seems to be an unrelated test failure, so maybe try merging with master to make sure that your branch is up-to-date

fourman-alex committed:
Merge branch 'upstream_master' into about_list_tile_fix_original

fourman-alex committed:
about_test.dart: formatting

shihaohong reviewed (approved):
LGTM
